presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/mysql/FunctionNamespaceDao.java

```diff
@@ -66,7 +67,7 @@
             "  create_time TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,\n" +
             "  update_time TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,\n" +
             "  PRIMARY KEY (id),\n" +
-            "  UNIQUE KEY function_id_version (function_id, version),\n" +
+            "  UNIQUE KEY function_id_hash_version (function_id_hash, version),\n" +
```

Comment:
Since this is a hash, what if there's collision?

Reply:
Writes would fail with a chance of 2^(-256), or 1e-77, which is just 0. User can modify any parts of the function (name, parameter name) to avoid collision.